### PR TITLE
resolved typescript problems for success.tsx, useTheme.tsx and event.ts

### PR DESCRIPTION
**Pull request by Heaust-ops**
Cleaned up some Typescript problems.

### CODE REVIEW

**Comment by PeerRich** [comment]
thank you! our team will look at it :)

File: pages/success.tsx [hunk 1]
```diff
@@ -24,8 +25,8 @@ dayjs.extend(toArray);
 dayjs.extend(timezone);
 
 export default function Success(props: inferSSRProps<typeof getServerSideProps>) {
-  const router = useRouter();
-  const { location, name, reschedule } = router.query;
+  const router: NextRouter = useRouter();
```
Comment (KATT) [inline pages/success.tsx:28]: This is inferred so it's obsolete

```suggestion
  const router = useRouter();
```

File: pages/success.tsx [hunk 2]
```diff
@@ -24,8 +25,8 @@ dayjs.extend(toArray);
...
+  const { location, name, reschedule }: ParsedUrlQuery = router.query;
```
Comment (KATT) [inline pages/success.tsx:29]: This is inferred so it's obsolete

```suggestion
  const { location, name, reschedule } = router.query;
```

File: pages/success.tsx [hunk 3]
```diff
@@ -141,11 +143,15 @@ export default function Success(props: inferSSRProps<typeof getServerSideProps>)
                               .utc()
                               .format("YYYYMMDDTHHmmss[Z]")}&text=${eventName}&details=${
                               props.eventType.description
-                            }` + (location ? "&location=" + encodeURIComponent(location) : "")
+                            }` +
+                            (location
+                              ? "&location=" +
+                                encodeURIComponent(typeof location === "string" ? location : false)
```
Comment (KATT) [inline pages/success.tsx:149]:
```suggestion
                                typeof location === "string" ? encodeURIComponent(location) : ''
```

File: pages/success.tsx [hunk 4]
```diff
@@ -240,14 +246,25 @@ export default function Success(props: inferSSRProps<typeof getServerSideProps>)
...
+    id: number;
+    title: string;
+    description: string | null;
+    length: number;
+    userId: number | null;
+    eventName: string | null;
+    requiresConfirmation: boolean;
+    team: {} | null; // todo: make an interface / type for team
+  }
+  const eventType: returnEventType | null = await prisma.eventType.findUnique({
```
Comment (KATT) [inline pages/success.tsx:267]: This is inferred so shouldn't be needed

```tsx
  const eventType = await prisma.eventType.findUnique({
```

Comment (KATT) [reply pages/success.tsx:267]: then follow-up with 

```ts
const eventTypeWithUsers = {
  ...eventType,
  users: !eventType.users.length && eventType.userId ? await prisma.user.findUnique({...}) : eventType.users`
}
```

**Review by KATT — CHANGES_REQUESTED** [review]
Thanks for this, the types overall in cal needs a lot of work. 🙏 

The thing I'd say overall is to try to lean on type inference over declared types, added comments inline.

File: pages/success.tsx (hunk 4 above)
Comment (KATT) [reply pages/success.tsx:267]: Or maybe that second part isn't needed, we can probably just `push()` it like we do now - the type inference should already guarantee that it'shas all the same fields

**Comment by Heaust-ops** [comment]
Made the requested changes : )
I'll always lean on inferred types now

**Review by emrysal — APPROVED** [review]
getServerSideProps(context: GetServerSidePropsContext) -> other than that all good! :)

File: pages/success.tsx [hunk 5]
```diff
@@ -240,14 +240,15 @@ export default function Success(props: inferSSRProps<typeof getServerSideProps>)
   );
 }
 
-export async function getServerSideProps(context) {
+export async function getServerSideProps(context: { query: { type: string } }) {
```
Comment (KATT) [inline pages/success.tsx:243]: as @emrysal  said
```suggestion
export async function getServerSideProps(context: GetServerSidePropsContext) {
```

**Comment by Heaust-ops** [comment]
Applied requested changes : )

**Pull request merged**